### ISSUE TITLE
Converting gpkg containing multiple images produces incorrect output

### ISSUE THREAD

**Issue by VirtualTim** [OSGeo/gdal#700]
Running gdal_translate (lossless output) on a gpkg containing multiple images introduces what looks like compression artefacts in the first image (first column).

I've attached an example in gpkg_bug.zip. raw_L.png and raw_R.png are the images extracted directly from the attached gpkg. raw_compiled.png is the two images manually stitched together. Running gdal_translate with a lossless output format I would expect the pixel values to be the same, however one of the images has some artefacts.

## Expected behavior and actual behavior.
The pixel values of output.fill and raw_compiled.png should be the same. As seen in "output_translate - output_expected.png" there is variation in the left most tile.

## Steps to reproduce the problem.
run> gdal_translate input.gpkg output.tiff
Compare output.tiff to raw_compiled.png (I used Beyond Compare)

## Operating system
For example: Windows 10 64 bit

## GDAL version and provenance
2.3.0 version from gisinternals (http://www.gisinternals.com/)

![output_translate <-> output_expected](https://user-images.githubusercontent.com/5424257/41632789-2c07693c-746e-11e8-8095-315fb5364a93.png)

[gpkg_bug.zip](https://github.com/OSGeo/gdal/files/2117458/gpkg_bug.zip)

**Comment by rouault** [comment]
OK, so your input.gpkg is in fact https://download.osgeo.org/gdal/data/geopackage/small_world_jpg_png.gpkg , which was generated with https://download.osgeo.org/gdal/data/geopackage/generate_gpkg_rasters.py

and the relevant comment is:
```
# 2x1 256x200 tiles. One with 3 bands in JPEG, the other one with 4 bands in PNG
os.system("gdal_translate -of GPKG small_world.tif small_world_jpg_png.gpkg -co BLOCKYSIZE=200")
```

The left tile being 256x200, the raster being 400x200, there is no transparency in it, so it can be compressed with JPEG. And that's the case when you dump the geopackage. The right tile being larger than the raster is encoded with PNG with transparency
So the result you observe is expected. The loss was introduced at GPKG creation, not during its output.

If you want lossless encoding, use -co TILE_FORMAT=PNG

**Comment by VirtualTim** [comment]
Ah you're right, I got my images mixed up. I've updated the bug with the correct data.

**Comment by VirtualTim** [comment]
@rouault I don't know if you get notified about comments on closed issues, so in case you missed it I have updated the bug with corrected info.

**Comment by rouault** [comment]
FWIW, on Linux, if I do:
```
gdal_translate small_world_jpg_png.rgba.400x200.gpkg out.tif -srcwin 0 0 256 200
gdal_translate small_world_jpg_png.rgba.400x200_linux.tiff out2.tif -srcwin 0 0 256 200
```
I get the same checksums with gdalinfo -checksum on out.tif, out2.tif and raw_L.JFIF (59265, 59826, 4169).
The difference between _linux and _windows must come from a different libjpeg version.
The maximum error between them is small:
```
$ gdalcompare.py small_world_jpg_png.rgba.400x200_linux.tiff small_world_jpg_png.rgba.400x200_windows.tiff 
Files differ at the binary level.
Band 1 checksum difference:
  Golden: 28263
  New:    28154
  Pixels Differing: 879
  Maximum Pixel Difference: 3.0
Band 2 checksum difference:
  Golden: 31136
  New:    31065
  Pixels Differing: 1062
  Maximum Pixel Difference: 2.0
Band 3 checksum difference:
  Golden: 36653
  New:    36565
  Pixels Differing: 900
  Maximum Pixel Difference: 3.0
Differences Found: 4
```
There is some tolerance/non-specified behaviour on JPEG decompression given this is a lossy compression (and you did your comparisons against raw_L.JFIF with Beyond compare which might use yet another libjpeg version or decompression settings)
So I don't think there is any bug.

**Comment by VirtualTim** [comment]
Alright. Thanks for taking the time to test this.

**Comment by VirtualTim** [comment]
This issue is fixed by #1378.
